### PR TITLE
Add alerting into Slack for build statuses

### PR DESCRIPTION
**Pull request by lennym**
Sends a slack message to #developers when a PR build fails or on any complete deployment irrespective of status.

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -47,3 +47,19 @@ jobs:
       - run: npm run build --if-present
       - run: npm start &
       - run: npm run test:integration
+
+  notify:
+    runs-on: ubuntu-latest
+    if: ${{ failure() }}
+    needs:
+      - test
+      - integrationtest
```
Comment (roc) [inline .github/workflows/build-and-test.yml:56]: Aside: If we have a `needs` context, can the alerter link to the failure?

File: .github/workflows/deploy-dev.yml [hunk 2]
```diff
@@ -119,3 +120,19 @@ jobs:
             --set image.tag=$IMAGE_TAG
             --set ckan.psql.initialize=false
             --set solr.initialize.enabled=false
+  notify:
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
```
Comment (roc) [inline .github/workflows/deploy-dev.yml:125]: Always for dev might get loud, though I guess let's start loud and tune down if we feel swamped 🙉

**Review by roc — APPROVED** [review]
🚨 whoop whoop

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (lennym) [reply .github/workflows/build-and-test.yml:56]: It does already. Not necessarily to the specific point of failure (there might be more than one), but to the Action run, from where it's fairly trivial to get to the point(s) of failure.

Screenshot 2022-04-07 at 10 50 15

Comment (roc) [reply .github/workflows/build-and-test.yml:56]: gorrit

**Pull request merged**